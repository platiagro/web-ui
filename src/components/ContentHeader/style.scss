.content-header {
  background: #fff;
  width: 100%;
}
