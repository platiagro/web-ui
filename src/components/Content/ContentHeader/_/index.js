// CORE LIBS
import React from 'react';
import PropTypes from 'prop-types';

// UI LIBS
import { PageHeader } from 'antd';
import { ArrowLeftOutlined } from '@ant-design/icons';

// COMPONENTS
import Title from '../../Title';

// STYLES
import './style.less';

/**
 * Content Header.
 * This component is responsible for displaying the content header with editable
 * title and go back arrow.
 *
 * @param props
 */
const ContentHeader = (props) => {
  // destructuring props
  const {
    handleGoBack,
    handleSubmit,
    title,
    loading,
    subTitle,
    extra,
    backIcon,
  } = props;

  // RENDER
  return (
    // page header component
    <PageHeader
      className='contentHeader'
      title={
        // title component
        <>
          <span className='subtitle-custom'>{subTitle}</span>
          <Title
            loading={loading}
            level={3}
            title={title}
            handleSubmit={handleSubmit}
          />
        </>
      }
      onBack={handleGoBack}
      extra={extra}
      backIcon={backIcon}
    />
  );
};

// PROP TYPES
ContentHeader.propTypes = {
  /** content header go back function */
  handleGoBack: PropTypes.func.isRequired,
  /** content header edit submit function */
  handleSubmit: PropTypes.func,
  /** content header title */
  title: PropTypes.string.isRequired,
  /** content header is loading */
  loading: PropTypes.bool.isRequired,
  /** if false the back icon will not be displayed */
  backIcon: PropTypes.bool.isRequired,
};

// PROP DEFAULT VALUES
ContentHeader.defaultProps = {
  /** content header edit submit function */
  handleSubmit: undefined,
  /* show back icon */
  backIcon: <ArrowLeftOutlined />,
};

// EXPORT
export default ContentHeader;
