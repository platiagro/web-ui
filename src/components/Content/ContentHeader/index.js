// CORE LIBS
import React from 'react';
import PropTypes from 'prop-types';

// UI LIBS
import { PageHeader } from 'antd';

// COMPONENTS
import Title from '../Title';

// STYLES
import './style.scss';

/**
 * Content Header.
 * This component is responsible for displaying the content header with editable
 * title and go back arrow.
 *
 * @param props
 */
const ContentHeader = (props) => {
  // destructuring props
  const { handleGoBack, handleSubmit, title, loading, subTitle, extra } = props;

  // RENDER
  return (
    // page header component
    <PageHeader
      className='contentHeader'
      title={
        // title component
        <>
          <span className='subtitle-custom'>{subTitle}</span>
          <Title
            loading={loading}
            level={3}
            title={title}
            handleSubmit={handleSubmit}
          />
        </>
      }
      onBack={handleGoBack}
      extra={
        <div style={{ marginTop: '10px', marginRight: '20px' }}>{extra}</div>
      }
    />
  );
};

// PROP TYPES
ContentHeader.propTypes = {
  /** content header go back function */
  handleGoBack: PropTypes.func.isRequired,
  /** content header edit submit function */
  handleSubmit: PropTypes.func,
  /** content header title */
  title: PropTypes.string.isRequired,
  /** content header is loading */
  loading: PropTypes.bool.isRequired,
};

// PROP DEFAULT VALUES
ContentHeader.defaultProps = {
  /** content header edit submit function */
  handleSubmit: undefined,
};

// EXPORT
export default ContentHeader;
