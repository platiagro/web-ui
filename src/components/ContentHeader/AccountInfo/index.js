import React from 'react';
import { UserOutlined } from '@ant-design/icons';

import './style.less';

const AccountInfo = () => (
  <div className='accountInfo'>
    <UserOutlined />
    <span>Usuário Anônimo</span>
  </div>
);

export default AccountInfo;
