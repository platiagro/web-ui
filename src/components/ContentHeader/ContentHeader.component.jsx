import React from 'react';
import { PageHeader } from 'antd';
import PropTypes from 'prop-types';
import { ArrowLeftOutlined } from '@ant-design/icons';

import Title from 'components/Title';

import './style.less';

const ContentHeader = ({
  title,
  extra,
  loading,
  subTitle,
  backIcon,
  customSubTitle,
  handleGoBack,
  handleSubmit,
}) => {
  return (
    <PageHeader
      className='contentHeader'
      onBack={handleGoBack}
      subTitle={subTitle}
      backIcon={backIcon}
      extra={extra}
      title={
        <>
          <span className='subtitle-custom'>{customSubTitle}</span>

          <Title
            level={3}
            title={title}
            loading={loading}
            handleSubmit={handleSubmit}
          />
        </>
      }
    />
  );
};

ContentHeader.propTypes = {
  title: PropTypes.string,
  extra: PropTypes.node,
  loading: PropTypes.bool,
  backIcon: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]),
  subTitle: PropTypes.string,
  customSubTitle: PropTypes.string,
  handleGoBack: PropTypes.func,
  handleSubmit: PropTypes.func,
};

ContentHeader.defaultProps = {
  title: undefined,
  extra: undefined,
  loading: false,
  backIcon: <ArrowLeftOutlined />,
  subTitle: undefined,
  customSubTitle: undefined,
  handleGoBack: undefined,
  handleSubmit: undefined,
};

export default ContentHeader;
