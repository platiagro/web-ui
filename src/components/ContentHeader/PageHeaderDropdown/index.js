// CORE LIBS
import React from 'react';
import { useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';

// UI LIBS
import { DownOutlined, ExperimentOutlined, ToolOutlined } from '@ant-design/icons';
import { Button, Dropdown, Menu } from 'antd';

// STYLES
import './style.less';

/**
 * Page Header Dropdown.
 * This component is responsible for page header dropdown to change view.
 *
 * @param {object} props Component props
 * 
 * @returns {PageHeaderDropdown} Component
 * 
 * @component
 */
const PageHeaderDropdown = (props) => {
  // destructuring props
  const {
    type,
    target
  } = props;

  let icon, title, targetTitle;

  switch (type) {
    case 'experiment':
      icon = <ExperimentOutlined />;
      title = 'Experimentação';
      targetTitle = 'Pré-implantação';
      break;
    case 'deployment':
      icon = <ToolOutlined />;
      title = 'Pré-implantação';
      targetTitle = 'Experimentação';
      break;
    default:
      break;
  }

  const changeView = () => {
    history.push(target);
  };

  const history = useHistory();

  const menu = (
    <Menu className='dropdown-menu'>
      <h1>Ir para:</h1>
      <Button
        shape='round'
        type='primary-inverse'
        size='medium'
        onClick={changeView}
      >
        {targetTitle}
      </Button>
    </Menu>
  );

  return (
    <div className='pageHeaderDropdown'>
      <Dropdown overlay={menu} trigger={['click']}>
        <Button
          className={type}
          type='primary-inverse'
          size='large'
          onClick={e => e.preventDefault()}
        >
          {icon}
          {title}
          <DownOutlined />
        </Button>
      </Dropdown>
    </div>
  )
}
// PROP TYPES
PageHeaderDropdown.propTypes = {
  /** Actual page type (experiment/deployment)*/
  type: PropTypes.string.isRequired,
  /** Target route for button*/
  target: PropTypes.string.isRequired
};

// EXPORT
export default PageHeaderDropdown;
