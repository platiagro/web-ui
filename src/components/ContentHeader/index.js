import React from 'react';
import PropTypes from 'prop-types';

import { PageHeader } from 'antd';

import './style.scss';

const ContentHeader = ({ title, subTitle }) => (
  <PageHeader className='content-header' title={title} subTitle={subTitle} />
);

ContentHeader.propTypes = {
  title: PropTypes.string,
  subTitle: PropTypes.string,
};

ContentHeader.defaultProps = {
  title: '',
  subTitle: '',
};

export default ContentHeader;
