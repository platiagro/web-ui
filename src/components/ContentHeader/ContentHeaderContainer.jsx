import React from 'react';
import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';

import ContentHeader from './index';
import AccountInfo from './AccountInfo';

const ContentHeaderContainer = ({
  title,
  subTitle,
  backIcon,
  customSubTitle,
}) => {
  const history = useHistory();

  const handleGoBack = () => {
    history.goBack();
  };

  return (
    <ContentHeader
      title={title}
      loading={false}
      subTitle={subTitle}
      backIcon={backIcon}
      extra={<AccountInfo />}
      handleGoBack={handleGoBack}
      customSubTitle={customSubTitle}
    />
  );
};

ContentHeaderContainer.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  backIcon: PropTypes.node,
  customSubTitle: PropTypes.string,
};

ContentHeaderContainer.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: undefined,
  backIcon: undefined,
  customSubTitle: undefined,
};

export default ContentHeaderContainer;
