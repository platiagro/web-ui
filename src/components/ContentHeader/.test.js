import React from 'react';

import { shallow } from 'enzyme';

import ContentHeader from '.';

describe('ContentHeader component should', () => {
  it('renders without crashing', () => {
    shallow(<ContentHeader />);
  });

  it('have a title props', () => {
    const contentHeaderShalowed = shallow(<ContentHeader />);

    expect(contentHeaderShalowed.prop('title')).toBeDefined();
  });

  it('have a subTitle props', () => {
    const contentHeaderShalowed = shallow(<ContentHeader />);

    expect(contentHeaderShalowed.prop('subTitle')).toBeDefined();
  });

  it('renders html correctly without title and subtitle', () => {
    const contentHeaderShalowed = shallow(<ContentHeader />);

    expect(contentHeaderShalowed.prop('title')).toBe('');
    expect(contentHeaderShalowed.prop('subTitle')).toBe('');
    expect(contentHeaderShalowed).toMatchSnapshot();
  });

  it('renders html correctly with title and no subtitle', () => {
    const title = 'Título';
    const contentHeaderShalowed = shallow(<ContentHeader title={title} />);

    expect(contentHeaderShalowed.prop('title')).toBe(title);
    expect(contentHeaderShalowed.prop('subTitle')).toBe('');
    expect(contentHeaderShalowed).toMatchSnapshot();
  });

  it('renders html correctly with subtitle and no title', () => {
    const subTitle = 'Subtítulo';
    const contentHeaderShalowed = shallow(
      <ContentHeader subTitle={subTitle} />
    );

    expect(contentHeaderShalowed.prop('title')).toBe('');
    expect(contentHeaderShalowed.prop('subTitle')).toBe(subTitle);
    expect(contentHeaderShalowed).toMatchSnapshot();
  });

  it('renders html correctly with title and subtitle', () => {
    const title = 'Título';
    const subTitle = 'Subtítulo';

    const contentHeaderShalowed = shallow(
      <ContentHeader title={title} subTitle={subTitle} />
    );

    expect(contentHeaderShalowed.prop('title')).toBe(title);
    expect(contentHeaderShalowed.prop('subTitle')).toBe(subTitle);
    expect(contentHeaderShalowed).toMatchSnapshot();
  });
});
