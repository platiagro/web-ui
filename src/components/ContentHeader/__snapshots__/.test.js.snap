// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentHeader component should renders html correctly with subtitle and no title 1`] = `
<PageHeader
  backIcon={
    <Icon
      type="arrow-left"
    />
  }
  className="content-header"
  subTitle="Subtítulo"
  title=""
/>
`;

exports[`ContentHeader component should renders html correctly with title and no subtitle 1`] = `
<PageHeader
  backIcon={
    <Icon
      type="arrow-left"
    />
  }
  className="content-header"
  subTitle=""
  title="Título"
/>
`;

exports[`ContentHeader component should renders html correctly with title and subtitle 1`] = `
<PageHeader
  backIcon={
    <Icon
      type="arrow-left"
    />
  }
  className="content-header"
  subTitle="Subtítulo"
  title="Título"
/>
`;

exports[`ContentHeader component should renders html correctly without title and subtitle 1`] = `
<PageHeader
  backIcon={
    <Icon
      type="arrow-left"
    />
  }
  className="content-header"
  subTitle=""
  title=""
/>
`;
